lib/puppet/util/log/destinations.rb

```diff
@@ -47,7 +47,11 @@ def handle(msg)
   require 'fileutils'
 
   def self.match?(obj)
-    Puppet::Util.absolute_path?(obj)
+    # TODO: what on earth is this supposed to be doing
```

Comment:
Need to spend some time understanding this code path... it's passing in `LogCollector` instances.

Reply:
It's trying to determine if this type of log destination should be used based on the specified log destination object. That way you can specify `--logdest console` vs `--logdest /path/to/log`, and the file based log destination will only match the latter. It's hacky but works given the logging API we have to work with.